narwhals/_duckdb/utils.py

```diff
@@ -120,6 +121,18 @@ def native_to_narwhals_dtype(duckdb_dtype: str, version: Version) -> DType:
         return dtypes.Boolean()
     if duckdb_dtype == "INTERVAL":
         return dtypes.Duration()
+    if duckdb_dtype.startswith("ENUM"):
+        if version is Version.V1:  # pragma: no cover
+            msg = "Converting to Enum is not supported in narwhals.stable.v1"
+            raise NotImplementedError(msg)
+
+        enum_content = re.search(r"ENUM\((.*?)\)", duckdb_dtype)
+        if not enum_content:  # pragma: no cover
+            msg = "Can not cast / initialize Enum without categories present"
+            raise ValueError(msg)
+        categories = re.findall(r"'(.*?)'", enum_content.group(1))
```

Comment:
TIL that duckdb types have a `children` attribute, which we could use instead of this to make it more robust
```python
In [30]: duckdb.sql("""SELECT 'happy'::ENUM ('sad', 'ok', 'happy')""").types[0]
Out[30]: ENUM('sad', 'ok', 'happy')

In [31]: duckdb.sql("""SELECT 'happy'::ENUM ('sad', 'ok', 'happy')""").types[0].children[0][1]
Out[31]: ['sad', 'ok', 'happy']
```
could we use that?